### PR TITLE
(visualstudiocode) Disable auto update by default

### PR DESCRIPTION
**Pull request by pascalberger**
Disable auto update by default. Added "EnableAutoUpdate" parameter for enable auto update

### CODE REVIEW

File: automatic/visualstudiocode/Tools/helpers.ps1 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            $storageFileContent.updateChannel = "none"
+            Write-Output "Updated 'updateChannel' to '$UpdateChannel'."
+        }
+        catch
+        {
+            Write-Output "Add new 'updateChannel' node with value '$UpdateChannel'."
+            $storageFileContent | Add-Member -Name "updateChannel" -value $UpdateChannel -MemberType NoteProperty
+        }
+
+        $storageFileContent | ConvertTo-Json -Depth 999 -Compress | Set-Content $storageFilePath
```
Comment (AdmiringWorm) [inline automatic/visualstudiocode/Tools/helpers.ps1]: using compress is wrong, I was just looking through the file and it is uncompressed

File: automatic/visualstudiocode/Tools/helpers.ps1 [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+{
+  "updateChannel": "$UpdateChannel"
+}
+"@
+
+        $storageFileContent | Set-Content $storageFilePath
+    } else {
+        Write-Output "Settings file '$storageFilePath' already exists."
+
+        $storageFileContent = ConvertFrom-Json "$(Get-Content $storageFilePath -Encoding UTF8)"
```
Comment (AdmiringWorm) [inline automatic/visualstudiocode/Tools/helpers.ps1]: You need to have a fallback for ConvertFrom-Json as I believe it isn't available in POSH v2

File: automatic/visualstudiocode/Tools/helpers.ps1 (hunk 1 above)
Comment (AdmiringWorm) [inline automatic/visualstudiocode/Tools/helpers.ps1]: also need to a fallback for ConvertTo-Json. I believe it isn't available in POSH v2

**Comment by AdmiringWorm** [comment]
hold off for now regarding the json convertto/from, I'm currently testing this package on a windows 7 base environment to see if it will work or not.
if it doesn't work, you could try the following from stackoverflow:
http://stackoverflow.com/a/29689642

it requires .net 3.5, but since .net 4.5.2 will still be installed, it shouldn't matter that much.

**Comment by AdmiringWorm** [comment]
@pascalberger so running this on a windows 7 vm throws an error.
relevant part:
```
2017-01-12 03:46:25,350 [INFO ] - Settings path 'C:\Users\vagrant\AppData\Roaming\Code' does not exist. Creating it now.
2017-01-12 03:46:25,374 [INFO ] - Settings file 'C:\Users\vagrant\AppData\Roaming\Code\storage.json' does not exist. Creating it now.
2017-01-12 03:46:25,387 [INFO ] - 
2017-01-12 03:46:25,387 [WARN ] - Type
2017-01-12 03:46:25,387 [WARN ] -   Confirmation (`-y`) is set.
  Respond within 30 seconds or the default selection will be chosen.
2017-01-12 03:46:52,106 [ERROR] - ERROR: Cannot process argument because the value of argument "type" is invalid. Change the value of the "type" argument and run the operation again.
```

**Comment by ferventcoder** [comment]
I disagree with disabling updates by default. Let's not make folks insecure in the name of having the right package version linked.

**Comment by pascalberger** [comment]
@ferventcoder Don't you want the functionallity at all or just keep auto update enabled by default and switch the parameter?

**Comment by ferventcoder** [comment]
Switch the parameter yes. Allow me to decide to remove the default behavior by adding something additional.

**Comment by pascalberger** [comment]
Thinking a little bit more about this, I would prefer to create a new manual package for just disabling auto update in Visual Studio Code instead of adding this functionallity as a parameter. Advantages of this would be that the disable-autoupdate package can be used as a dependency within another package for setting up an environment. And it keeps thethe Visual Studio Code package clean by not adding functionality to the package which is not part of their installer experience. And we can still mention the new package for disabling auto update in the package description of the Visual Studio Code package so that people easily can find it.

@AdmiringWorm @ferventcoder Any thoughts and preferences about this?

**Comment by gep13** [comment]
@pascalberger I like the idea of a separate package.  Seems cleaner to me.

**Comment by gep13** [comment]
Sorry, didn't mean to close the PR 😄

**Comment by ferventcoder** [comment]
:+1: for separate package!

**Comment by pascalberger** [comment]
Changed functionallity to disable auto-updater to its own package. 

@AdmiringWorm Should be ready for another review

**Comment by AdmiringWorm** [comment]
@pascalberger I'll have a look when I get home (around 5-6 hours from now)

**Comment by ferventcoder** [comment]
Looks good. I didn't check the PowerShell code in detail though.

**Pull request merged**